### ISSUE TITLE
Missing SSL cert with SSL off

### ISSUE THREAD

**Issue by franz-josef-kaiser** [puphpet/puphpet#1073]
On the online config `https://puphpet.com/upload-config#webserver`, I have SSL _not checked_. Still, when I try to check on Apache2 on the command line

    apache2 -t

I get the following errors:

    [Tue Sep 02 14:08:31.296894 2014] [core:warn] [pid 23031] 
    AH00111: Config variable ${APACHE_PID_FILE} is not defined
    [Tue Sep 02 14:08:31.349619 2014] [core:warn] [pid 23031:tid 140117853153152] 
    AH00111: Config variable ${APACHE_RUN_DIR} is not defined
    [Tue Sep 02 14:08:31.353290 2014] [core:warn] [pid 23031:tid 140117853153152] 
    AH00111: Config variable ${APACHE_LOCK_DIR} is not defined
    [Tue Sep 02 14:08:31.361104 2014] [core:warn] [pid 23031:tid 140117853153152]
    AH00111: Config variable ${APACHE_RUN_DIR} is not defined

    AH00526: Syntax error on line 36 of /etc/apache2/sites-enabled/10-default_vhost_443.conf:
    SSLCertificateKeyFile: file '/etc/ssl/private/ssl-cert-snakeoil.key' does not exist or is empty

When I check for the `.conf` file

    sudo nano /etc/apache2/sites-enabled/10-default_vhost_443.conf

I find the following directives causing the error:

    ## SSL directives
    SSLEngine on
    SSLCertificateFile      "/etc/ssl/certs/ssl-cert-snakeoil.pem"
    SSLCertificateKeyFile   "/etc/ssl/private/ssl-cert-snakeoil.key"
    SSLCACertificatePath    "/etc/ssl/certs"

I can leave the `SSLEngine on`, but I have to comment the certs and path out to get rid of that "Syntax" error. The **problem** with that approach is, that then Apache2 won't restart. From my log file:

    [Tue Sep 02 14:02:59.491178 2014] [ssl:emerg] [pid 22825:tid 140019971700608] 
    AH02311: Fatal error initialising mod_ssl, exiting. 
    See /var/log/apache2/default_vhost_44$AH00016: Configuration Failed

_broken up into sep. lines for readability._

The shortened `config.yaml`:

```
---
vagrantfile-local:
    vm:
        box: puphpet/ubuntu1404-x64
        box_url: puphpet/ubuntu1404-x64
        hostname: ''
        memory: '512'
        cpus: '1'
        chosen_provider: virtualbox
        network:
            private_network: 192.168.56.101
            forwarded_port:
                VbUQLGZlh1HC:
                    host: '8080'
                    guest: '22'
        post_up_message: ''
        provider:
            virtualbox:
                modifyvm:
                    natdnshostresolver1: on
            vmware:
                numvcpus: 1
            parallels:
                cpus: 1
        provision:
            puppet:
                manifests_path: puphpet/puppet
                manifest_file: site.pp
                module_path: puphpet/puppet/modules
                options:
                    - '--verbose'
                    - '--hiera_config /vagrant/puphpet/puppet/hiera.yaml'
                    - '--parser future'
        synced_folder:
            pUNWnuwo76ip:
                source: ./
                target: /var/www
                sync_type: default
                rsync:
                    args:
                        - '--verbose'
                        - '--archive'
                        - '-z'
                    exclude:
                        - .vagrant/
                    auto: 'false'
        usable_port_range:
            start: 10200
            stop: 10500
    ssh:
        host: null
        port: null
        private_key_path: null
        username: vagrant
        guest_port: null
        keep_alive: true
        forward_agent: false
        forward_x11: false
        shell: 'bash -l'
    vagrant:
        host: detect
(...)
apache:
    install: '1'
    settings:
        user: www-data
        group: www-data
        default_vhost: true
        manage_user: false
        manage_group: false
        sendfile: 0
    modules:
        - rewrite
    vhosts:
        mQ8mXHIlJigP:
            servername: dumbair.dev
            serveraliases:
                - www.dumbair.dev
            docroot: /var/www/dumbair
            port: '80'
            setenv:
                - 'APP_ENV dev'
            override:
                - All
            options:
                - Indexes
                - FollowSymLinks
                - MultiViews
            engine: php
            custom_fragment: ''
            ssl_cert: ''
            ssl_key: ''
            ssl_chain: ''
            ssl_certs_dir: ''
    mod_pagespeed: 0
nginx:
    install: '0'
    settings:
        default_vhost: 1
        proxy_buffer_size: 128k
        proxy_buffers: '4 256k'
    vhosts:
        VA2j6oIAnAxe:
            server_name: awesome.dev
            server_aliases:
                - www.awesome.dev
            www_root: /var/www/awesome
            listen_port: '80'
            location: \.php$
            index_files:
                - index.html
                - index.htm
                - index.php
            envvars:
                - 'APP_ENV dev'
            engine: php
            ssl_cert: ''
            ssl_key: ''
(...)
```

I can see `mod_ssl` enabled inside `/etc/apache2/mods-enabled/ssl.conf` with the following contents:

```
<IfModule mod_ssl.c>
  SSLRandomSeed startup builtinev/urandom 512
  SSLRandomSeed startup file:/dev/urandom 512
  SSLRandomSeed connect builtinev/urandom 512
  SSLRandomSeed connect file:/dev/urandom 512

  AddType application/x-x509-ca-cert .crt
  AddType application/x-pkcs7-crl    .crl

  SSLPassPhraseDialog  builtin
  SSLSessionCache "shmcb:${APACHE_RUN_DIR}/ssl_scache(512000)"
  SSLSessionCacheTimeout 300
    Mutex default
    SSLCryptoDevice builtin
  SSLHonorCipherOrder On
  SSLCipherSuite HIGH:MEDIUM:!aNULL:!MD5
  SSLProtocol all -SSLv2
  SSLOptions StdEnvVars
</IfModule>
```

And from `/etc/apache2/modes-enabled/ssl.load`:

    LoadModule ssl_module /us /lib/apache2/modules/mod_ssl.so

**Comment by caseycs** [comment]
Try `default_vhost: false` in the `apache:` section. Worked for me.

**Comment by jtreminio** [comment]
Fixed by 3b2fca221e37a7dd8abd21457f634453de2418b5
